### ISSUE TITLE
Not wanted tags __shape and __coordinates in piece XML

### ISSUE THREAD

**Issue by SKoschnicke** [CAU-Kiel-Tech-Inf/backend#351]
Since version 21.2.1 there are tags `__shape` and `__coordinates` in the `piece` XML (lastMove).

These shouldn't be there.

**Comment by xerus2000** [comment]
Seems fixed

**Comment by SKoschnicke** [comment]
They seem to be still present in current master (8d2c03bf):

<details>

```xml
<protocol>
  <join gameType="swc_2021_blokus"/><protocol>
  <joined roomId="67575a73-537a-4cab-8d7a-f5bfc42ee824"/>
  <room roomId="67575a73-537a-4cab-8d7a-f5bfc42ee824">
    <data class="welcomeMessage" color="ONE"/>
  </room>
  <room roomId="67575a73-537a-4cab-8d7a-f5bfc42ee824">
    <data class="memento">
      <state class="state" turn="0" round="1" startPiece="PENTO_P">
        <startTeam class="team">ONE</startTeam>
        <blueShapes class="linked-hash-set">
          <shape>MONO</shape>
          <shape>DOMINO</shape>
          <shape>TRIO_L</shape>
          <shape>TRIO_I</shape>
          <shape>TETRO_O</shape>
          <shape>TETRO_T</shape>
          <shape>TETRO_I</shape>
          <shape>TETRO_L</shape>
          <shape>TETRO_Z</shape>
          <shape>PENTO_L</shape>
          <shape>PENTO_T</shape>
          <shape>PENTO_V</shape>
          <shape>PENTO_S</shape>
          <shape>PENTO_Z</shape>
          <shape>PENTO_I</shape>
          <shape>PENTO_P</shape>
          <shape>PENTO_W</shape>
          <shape>PENTO_U</shape>
          <shape>PENTO_R</shape>
          <shape>PENTO_X</shape>
          <shape>PENTO_Y</shape>
        </blueShapes>
        <yellowShapes class="linked-hash-set">
          <shape>MONO</shape>
          <shape>DOMINO</shape>
          <shape>TRIO_L</shape>
          <shape>TRIO_I</shape>
          <shape>TETRO_O</shape>
          <shape>TETRO_T</shape>
          <shape>TETRO_I</shape>
          <shape>TETRO_L</shape>
          <shape>TETRO_Z</shape>
          <shape>PENTO_L</shape>
          <shape>PENTO_T</shape>
          <shape>PENTO_V</shape>
          <shape>PENTO_S</shape>
          <shape>PENTO_Z</shape>
          <shape>PENTO_I</shape>
          <shape>PENTO_P</shape>
          <shape>PENTO_W</shape>
          <shape>PENTO_U</shape>
          <shape>PENTO_R</shape>
          <shape>PENTO_X</shape>
          <shape>PENTO_Y</shape>
        </yellowShapes>
        <redShapes class="linked-hash-set">
          <shape>MONO</shape>
          <shape>DOMINO</shape>
          <shape>TRIO_L</shape>
          <shape>TRIO_I</shape>
          <shape>TETRO_O</shape>
          <shape>TETRO_T</shape>
          <shape>TETRO_I</shape>
          <shape>TETRO_L</shape>
          <shape>TETRO_Z</shape>
          <shape>PENTO_L</shape>
          <shape>PENTO_T</shape>
          <shape>PENTO_V</shape>
          <shape>PENTO_S</shape>
          <shape>PENTO_Z</shape>
          <shape>PENTO_I</shape>
          <shape>PENTO_P</shape>
          <shape>PENTO_W</shape>
          <shape>PENTO_U</shape>
          <shape>PENTO_R</shape>
          <shape>PENTO_X</shape>
          <shape>PENTO_Y</shape>
        </redShapes>
        <greenShapes class="linked-hash-set">
          <shape>MONO</shape>
          <shape>DOMINO</shape>
          <shape>TRIO_L</shape>
          <shape>TRIO_I</shape>
          <shape>TETRO_O</shape>
          <shape>TETRO_T</shape>
          <shape>TETRO_I</shape>
          <shape>TETRO_L</shape>
          <shape>TETRO_Z</shape>
          <shape>PENTO_L</shape>
          <shape>PENTO_T</shape>
          <shape>PENTO_V</shape>
          <shape>PENTO_S</shape>
          <shape>PENTO_Z</shape>
          <shape>PENTO_I</shape>
          <shape>PENTO_P</shape>
          <shape>PENTO_W</shape>
          <shape>PENTO_U</shape>
          <shape>PENTO_R</shape>
          <shape>PENTO_X</shape>
          <shape>PENTO_Y</shape>
        </greenShapes>
        <validColors class="linked-hash-set">
          <color>BLUE</color>
          <color>YELLOW</color>
          <color>RED</color>
          <color>GREEN</color>
        </validColors>
        <first displayName="Unknown">
          <color class="team">ONE</color>
        </first>
        <second displayName="Unknown">
          <color class="team">TWO</color>
        </second>
        <board/>
        <lastMoveMono class="linked-hash-map"/>
      </state>
    </data>
  </room>
  <room roomId="67575a73-537a-4cab-8d7a-f5bfc42ee824">
    <data class="sc.framework.plugins.protocol.MoveRequest"/>
  </room>
  <room roomId="67575a73-537a-4cab-8d7a-f5bfc42ee824">
    <data class="sc.plugin2021.SetMove">
      <piece color="BLUE" kind="PENTO_P" rotation="MIRROR" isFlipped="true">
        <__shape class="linked-hash-set">
          <coordinates x="0" y="2"/>
          <coordinates x="1" y="2"/>
          <coordinates x="0" y="1"/>
          <coordinates x="1" y="1"/>
          <coordinates x="0" y="0"/>
        </__shape>
        <__coordinates class="linked-hash-set">
          <coordinates x="0" y="19"/>
          <coordinates x="1" y="19"/>
          <coordinates x="0" y="18"/>
          <coordinates x="1" y="18"/>
          <coordinates x="0" y="17"/>
        </__coordinates>
        <position x="0" y="17"/>
      </piece>
    </data>
  </room>
  <room roomId="67575a73-537a-4cab-8d7a-f5bfc42ee824">
    <data class="memento">
      <state class="state" turn="1" round="1" startPiece="PENTO_P">
        <startTeam class="team">ONE</startTeam>
        <blueShapes class="linked-hash-set">
          <shape>MONO</shape>
          <shape>DOMINO</shape>
          <shape>TRIO_L</shape>
          <shape>TRIO_I</shape>
          <shape>TETRO_O</shape>
          <shape>TETRO_T</shape>
          <shape>TETRO_I</shape>
          <shape>TETRO_L</shape>
          <shape>TETRO_Z</shape>
          <shape>PENTO_L</shape>
          <shape>PENTO_T</shape>
          <shape>PENTO_V</shape>
          <shape>PENTO_S</shape>
          <shape>PENTO_Z</shape>
          <shape>PENTO_I</shape>
          <shape>PENTO_W</shape>
          <shape>PENTO_U</shape>
          <shape>PENTO_R</shape>
          <shape>PENTO_X</shape>
          <shape>PENTO_Y</shape>
        </blueShapes>
        <yellowShapes class="linked-hash-set">
          <shape>MONO</shape>
          <shape>DOMINO</shape>
          <shape>TRIO_L</shape>
          <shape>TRIO_I</shape>
          <shape>TETRO_O</shape>
          <shape>TETRO_T</shape>
          <shape>TETRO_I</shape>
          <shape>TETRO_L</shape>
          <shape>TETRO_Z</shape>
          <shape>PENTO_L</shape>
          <shape>PENTO_T</shape>
          <shape>PENTO_V</shape>
          <shape>PENTO_S</shape>
          <shape>PENTO_Z</shape>
          <shape>PENTO_I</shape>
          <shape>PENTO_P</shape>
          <shape>PENTO_W</shape>
          <shape>PENTO_U</shape>
          <shape>PENTO_R</shape>
          <shape>PENTO_X</shape>
          <shape>PENTO_Y</shape>
        </yellowShapes>
        <redShapes class="linked-hash-set">
          <shape>MONO</shape>
          <shape>DOMINO</shape>
          <shape>TRIO_L</shape>
          <shape>TRIO_I</shape>
          <shape>TETRO_O</shape>
          <shape>TETRO_T</shape>
          <shape>TETRO_I</shape>
          <shape>TETRO_L</shape>
          <shape>TETRO_Z</shape>
          <shape>PENTO_L</shape>
          <shape>PENTO_T</shape>
          <shape>PENTO_V</shape>
          <shape>PENTO_S</shape>
          <shape>PENTO_Z</shape>
          <shape>PENTO_I</shape>
          <shape>PENTO_P</shape>
          <shape>PENTO_W</shape>
          <shape>PENTO_U</shape>
          <shape>PENTO_R</shape>
          <shape>PENTO_X</shape>
          <shape>PENTO_Y</shape>
        </redShapes>
        <greenShapes class="linked-hash-set">
          <shape>MONO</shape>
          <shape>DOMINO</shape>
          <shape>TRIO_L</shape>
          <shape>TRIO_I</shape>
          <shape>TETRO_O</shape>
          <shape>TETRO_T</shape>
          <shape>TETRO_I</shape>
          <shape>TETRO_L</shape>
          <shape>TETRO_Z</shape>
          <shape>PENTO_L</shape>
          <shape>PENTO_T</shape>
          <shape>PENTO_V</shape>
          <shape>PENTO_S</shape>
          <shape>PENTO_Z</shape>
          <shape>PENTO_I</shape>
          <shape>PENTO_P</shape>
          <shape>PENTO_W</shape>
          <shape>PENTO_U</shape>
          <shape>PENTO_R</shape>
          <shape>PENTO_X</shape>
          <shape>PENTO_Y</shape>
        </greenShapes>
        <validColors class="linked-hash-set">
          <color>BLUE</color>
          <color>YELLOW</color>
          <color>RED</color>
          <color>GREEN</color>
        </validColors>
        <first displayName="Unknown">
          <color class="team">ONE</color>
        </first>
        <second displayName="Unknown">
          <color class="team">TWO</color>
        </second>
        <board>
          <field x="0" y="17" content="BLUE"/>
          <field x="0" y="18" content="BLUE"/>
          <field x="1" y="18" content="BLUE"/>
          <field x="0" y="19" content="BLUE"/>
          <field x="1" y="19" content="BLUE"/>
        </board>
        <lastMove class="sc.plugin2021.SetMove">
          <piece color="BLUE" kind="PENTO_P" rotation="MIRROR" isFlipped="true">
            <__shape class="linked-hash-set">
              <coordinates x="0" y="2"/>
              <coordinates x="1" y="2"/>
              <coordinates x="0" y="1"/>
              <coordinates x="1" y="1"/>
              <coordinates x="0" y="0"/>
            </__shape>
            <__coordinates class="linked-hash-set">
              <coordinates x="0" y="19"/>
              <coordinates x="1" y="19"/>
              <coordinates x="0" y="18"/>
              <coordinates x="1" y="18"/>
              <coordinates x="0" y="17"/>
            </__coordinates>
            <position x="0" y="17"/>
          </piece>
        </lastMove>
        <lastMoveMono class="linked-hash-map"/>
      </state>
    </data>
  </room>
```

**Comment by xerus2000** [comment]
Interesting, I tested a serialisation of `lastMove` in https://github.com/CAU-Kiel-Tech-Inf/backend/pull/374/files#diff-5e18167ae19f8b1b6b2d539af6baebf5522cacc68c18402830d2f6c90b7b9257R176-R180 and it worked - but that may be related to not using some lazy fields.

**Comment by xerus2000** [comment]
Fixed in 23589a153e8cd3c5b1b3257ff35f66ebbb8d3012
